### PR TITLE
test: ensure 'InaccessiblePaths=/proc' option works

### PR DESCRIPTION
**Pull request by Siosm**
Test case for PR #5985.

### CODE REVIEW

**Comment by Siosm** [comment]
Apparently, this test is skipped by CI (needs root).

File: test/test-execute/exec-inaccessiblepaths-proc.service [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[Unit]
+Description=Test to make sure that mount namespace setup works properly with the 'InaccessiblePaths=/proc' option
+
+[Service]
+InaccessiblePaths=/proc
+ExecStart=/bin/sh -x -c 'test ! "$(ls -A /proc)"'
```
Comment (evverx) [inline test/test-execute/exec-inaccessiblepaths-proc.service]: Thank you for adding the test.

I'm not sure I understand why we need `-A`. Besides, I think the check is a bit complicated. Probably we should simply check that the access rights are `0`.

Talking of running `test-execute` by the CI, https://anonscm.debian.org/cgit/pkg-systemd/systemd.git/tree/debian/tests/root-unittests runs it.

Comment (evverx) [reply test/test-execute/exec-inaccessiblepaths-proc.service]: Also, we always double `$`. cdaf5070480c0 explains why we do that.

**Comment by Siosm** [comment]
Thanks for the review. Updated.

**Comment by evverx** [comment]
I'm sorry I forgot to mention that we also double `%`.

**Comment by Siosm** [comment]
My bad. Fixed.

**Pull request merged**